# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/TestingFunctions.cpp
# Commit: 3c25ee26a7ad
# Full Hash: 3c25ee26a7adab3e323b7f0409a7b4dca2f0f9f8
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2024-10-15 04:16:11
# Description:
#   Bug 1924062 - Throw exception for asm.js module functions in disnative testing function. r=arai
#   
#   A function marked `isAsmJSNative` is either the outer module function or an actual
#   exported asm.js function. The module function can't be disassembled.
#   
# ==============================================================================

diff -r 3613fc672c50 -r 3c25ee26a7ad js/src/builtin/TestingFunctions.cpp
--- a/js/src/builtin/TestingFunctions.cpp	Mon Oct 14 10:32:25 2024 +0000
+++ b/js/src/builtin/TestingFunctions.cpp	Mon Oct 14 10:38:58 2024 +0000
@@ -1780,6 +1780,10 @@
   uint8_t* jit_end = nullptr;
 
   if (fun->isAsmJSNative() || fun->isWasmWithJitEntry()) {
+    if (IsAsmJSModule(fun)) {
+      JS_ReportErrorASCII(cx, "Can't disassemble asm.js module function.");
+      return false;
+    }
     if (fun->isAsmJSNative()) {
       sprinter.printf("; backend=asmjs\n");
     }